src/realm-settings/NewClientPolicyForm.tsx

```diff
@@ -143,29 +145,37 @@ export default function NewClientPolicyForm() {
       "org.keycloak.services.clientpolicy.condition.ClientPolicyConditionProvider"
     ];
 
-  const formValues = form.getValues();
-
   const save = async () => {
-    const createdForm = formValues;
+    const createdForm = form.getValues();
     const createdPolicy = {
       ...createdForm,
       profiles: [],
       conditions: [],
     };
 
-    const policyNameExists = policies.find(
-      (policy) => policy.name === createdPolicy.name
-    );
+    const getAllPolicies = () => {
+      const policyNameExists = !!policies.find(
+        (policy) => policy.name === createdPolicy.name
+      );
 
-    const res = policies.map((policy) =>
-      policy.name === createdPolicy.name ? createdPolicy : policy
-    );
+      const res = policies.map((policy) =>
+        policy.name === createdPolicy.name ? createdPolicy : policy
+      );
 
-    const allPolicies = policyNameExists ? res : policies.concat(createdForm);
+      if (policyNameExists) {
+        return res;
+      } else if (createdForm.name !== policyName) {
+        return policies
+          .filter((item) => item.name !== policyName)
+          .concat(createdForm);
+      } else {
+        return policies.concat(createdForm);
+      }
```

Comment:
Since all the branches of this `if` statement return a value you can omit the `else` clause and return directly.
```suggestion
      return policies.concat(createdForm);
```